### PR TITLE
Bug fix/optimizations 06 08 2019

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Copy less data, especially in ClusterInfo::loadPlan()
* Automatically turn-off smart collections in Community mode

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all cluster tests*.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5563/

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -925,7 +920,7 @@ void ClusterInfo::loadPlan() {
         }
       }
 
-      newCollections.emplace(std::make_pair(databaseName, databaseCollections));
+      newCollections.emplace(std::make_pair(std::move(databaseName), std::move(databaseCollections)));
```
Comment (graetzer) [inline arangod/Cluster/ClusterInfo.cpp]: why is make pair necessary here ?

File: arangod/Agency/Agent.cpp [hunk 2]
```diff
@@ -1200,14 +1200,15 @@ read_ret_t Agent::read(query_t const& query) {
     return read_ret_t(false, NO_LEADER);
   }
 
-  std::string leaderId = _constituent.leaderID();
+  leader = _constituent.leaderID();
```
Comment (maierlars) [inline arangod/Agency/Agent.cpp]: why do we have to relead `leader` at all?

Comment (jsteemann) [reply arangod/Agency/Agent.cpp]: That re-reading of the leader was introduced in 22e46978a6ce38d69fd5519188824b7e2dd67037 by @neunhoef. Please check with him.

File: arangod/Cluster/ClusterInfo.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Cluster/ClusterInfo.cpp]: It's not necessary. Will remove it

**Comment by jsteemann** [comment]
Re-test after changes:
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5567/

**Pull request merged**